### ISSUE TITLE
remove need to restart Processing to see new sketches in the sketchbook folder

### ISSUE THREAD

**Issue by JobLeonard** [processing/processing#3147]
Right now, if I change anything in the sketchbook folder, like copying a new sketch or removing them, I have to close and restart the PDE before it notices the change. 

Assuming there is some kind of scan-directory code that is run once when the PDE starts, it would be ice if that could instead be run whenever the sketchbook window is opened.

Alternatively, perhaps add a refresh button in the sketchbook window?

**Comment by Akarshit** [comment]
Very close to #2944

**Comment by JobLeonard** [comment]
Apologies, I did a quick search for "sketchbook" in the issues expecting this issue has been brought up many times before (rightfully so, it seems). And somehow I read over all of these.

Having said that, this has caused real confusion among my students! Also frustration among the more tech-savvy: sometimes they have multiple sketches open to take algorithms from for their own little project, and they have to close all of that to open the new sketch (or use the OPEN command, but that has the downside that it doesn't default to the sketchbook folder, so they have to navigate their file tree every time they use it - actually, can I open an issue for that or does that already exist as well?)

**Comment by Akarshit** [comment]
What do you mean by 
`they have to close all of that to open the new sketch`  
Feel free to open issues . If any of then is a duplicate they will automatically be noticed by some member of the community.

**Comment by JobLeonard** [comment]
Sorry, I worded that wrong - the OPEN functionality works fine (except for the part where it doesn't default to the sketchbook folder, IMO, but that might be a conscious decision). What I intended to say was:

- Students have many sketches open, to mix and match solutions from them (one about colors, one about bouncing objects, etcetera)
- They share a sketch with each other as zip files
- They unzip the sketch to the sketchbook folder
- They then either have the choice of
 - using OPEN, then hunt for the Processing sketchbook, or 
 - close everything and restart Processing, then open everything again

It's a very minor inconvenience, I admit, but if the sketchbook is auto-updated it would simplify things.

Actually, do *open* sketches auto-update their view if their ````.pde```` files are externally modified? If so, students could use a shared dropbox folder and work on the same sketch in different tabs, for example (although at that stage they're better off using proper version control, of course).
